File: app/src/main/java/com/example/app/ApplicationExt.java
package com.example.app;

import android.app.Application;

public class ApplicationExt extends Application {
    private MyComponent mMyComponent;
    @Override
    public void onCreate() {
        super.onCreate();
        mMyComponent = createMyComponent();
    }

    public MyComponent getMyComponent() {
        return mMyComponent;
    }

    private MyComponent createMyComponent() {
        return DaggerMyComponent
                .builder()
                .myModule(new MyModule())
                .build();
    }

}

File: app/src/main/java/com/example/app/ApplicationExt.kt
package com.example.app

import android.app.Application
import dagger.internal.DaggerCollections

class ApplicationExt : Application() {
    var myComponent: MyComponent? = null
        private set

    override fun onCreate() {
        super.onCreate()
//        myComponent = createMyComponent()
    }

    private fun createMyComponent(): MyComponent {
//        return DaggerMyComponent.builder()
//                .myModule(MyModule())
//                .build()
        return createMyComponent()
    }

}
